src/server/main.py

```diff
@@ -0,0 +1,370 @@
+        :param load_dotenv: Load the nearest :file:`.env` and :file:`.flaskenv`
+            files to set environment variables. Will also change the working
+            directory to the directory containing the first file found.
+        :param options: the options to be forwarded to the underlying Werkzeug
+            server. See :func:`werkzeug.serving.run_simple` for more
+            information.
+        .. versionchanged:: 1.0
+            If installed, python-dotenv will be used to load environment
+            variables from :file:`.env` and :file:`.flaskenv` files.
+            The :envvar:`FLASK_DEBUG` environment variable will override :attr:`debug`.
+            Threaded mode is enabled by default.
+        .. versionchanged:: 0.10
+            The default port is now picked from the ``SERVER_NAME``
+            variable.
+
+        """
+        self.app.run(**kwargs)
+
+    def route_run(self):
+        """
+        Handle the start of an instance.
+
+        expected form :
+
+        {
+            "type":"drl|ga",
+            "params":{
+                "nb_epochs":"int",
+                "pop_size?":"int",
+                "mutation_rate?":"float",
+                "crossover_rate?":"float",
+                "learning_rate?":"float",
+                "batch_size?":"int",
+                "momentum?":"float"
+            },
+            "override":"bool",
+            "bench_id":"string",
+            "snapshot_rate":"int",
+            "seed":"int"
+        }
+
+        Returns :
+        {
+            “busy”:”bool”
+        }
+
+
+        """
+        if self._runner is not None:
+            return {"busy": True}
+
+        if request.method == "POST":
+            metho = request.form["type"]
+            if metho not in HYPER_LIST:
+                raise TypeError(f"Unkown method {metho}")
+            param = json.loads(request.form["param"])
+            hyper = {key: param[key] for key in HYPER_LIST[metho]}
+
+            self._nb_it = hyper["nb_epochs"]
+            self._count = 0
+
+            self._rate = int(request.form["snapshot_rate"])
+
+            bench = utils.create_ecvrp(utils.parse_dataset(request.form["bench_id"]))
+
+            random.seed(int(request.form["seed"]))
+
+            self._name = f"{request.form['bench_id']}_{metho}_{request.form['seed']}"
+            for param in hyper.values():
+                self._name += f"_{param}"
+
+            if metho == "ga":
+                g_a = GA(
+                        build_first_gen(hyper["pop_size"], bench),
+                        hyper["mutation_rate"],
+                        request.form["seed"]
+                    )
+                self._runner = g_a.run(hyper["nb_epochs"])
+                self._tot_time = 0
+
+            self._override = bool(request.form["override"])
+
+            self._snapshot = JsonWriter(
+                str(utils.PATH_TO_LOGS),
+                self._name,
+                request.form["bench_id"],
```

Comment:
I consider it as the name of the benchmark because that's what has been accepted in the original documentation. The bench_id is a string. I also would like to add that as stated in a previous conversation working with indexes in this case could lead to problems. If a benchmark is added or removed some indexes might be shifted while the the names would stay.

Reply:
Ok. I will change that, it will be quick.